### PR TITLE
Additional vc_asserts

### LINKED ISSUE #892: Add `vc_assert MOBILE_EVENT_OBJECT_GS_BALL == $B` for `vc_hook Enable_GS_Ball_mobile_event`
This hook is hard-coded in the VC emulator to write $0b to $01:be3c and $01:be44. We assert that those locations are `sMobileEventIndex` and `sMobileEventIndexBackup`, but do not yet assert that the value is `MOBILE_EVENT_OBJECT_GS_BALL`.

### PR DESCRIPTION
**Pull request by vulcandth**
This PR adds additional Virtual Console build asserts.

Please reference #892 and Rangi42/polishedcrystal#684

Resolves #892

### CODE REVIEW

File: engine/menus/save.asm [hunk 1]
```diff
@@ -170,6 +170,8 @@ AddHallOfFameEntry:
 		"sMobileEventIndex is no longer located at 01:be3c."
 	vc_assert BANK(sMobileEventIndexBackup) == $1 && sMobileEventIndexBackup == $be44, \
 		"sMobileEventIndexBackup is no longer located at 01:be44."
+	vc_assert MOBILE_EVENT_OBJECT_GS_BALL == $B, \
+		"MOBILE_EVENT_OBJECT_GS_BALL is no longer equal to $B."
```
Comment (Rangi42) [inline engine/menus/save.asm:174]:
```suggestion
	vc_assert MOBILE_EVENT_OBJECT_GS_BALL == $0b, \
		"MOBILE_EVENT_OBJECT_GS_BALL is no longer equal to $0b."
```

File: engine/link/link.asm [hunk 2]
```diff
@@ -2311,6 +2311,14 @@ WaitForLinkedFriend:
 	ldh [rSC], a
 	ld a, (1 << rSC_ON) | (0 << rSC_CLOCK)
 	vc_hook linkCable_fake_begin
```
Comment (Rangi42) [inline engine/link/link.asm:2313]: Add a comment above this `vc_hook` like the Celebi one, explaining briefly how it uses these values.

**Comment by vulcandth** [comment]
Going to do a little bit extra testing real quick to see If I can't be a bit more specific on what the linkCable\_fake hooks are doing.

**Comment by vulcandth** [comment]
Removed two of the vc_assert's that I determined were not being used through testing. Added the one comment you requested. 

Further patch Identification and Comments will be done in the other PR.

File: engine/link/link.asm [hunk 3]
```diff
@@ -2310,7 +2310,14 @@ WaitForLinkedFriend:
 	ld a, (0 << rSC_ON) | (0 << rSC_CLOCK)
 	ldh [rSC], a
 	ld a, (1 << rSC_ON) | (0 << rSC_CLOCK)
+; This vc_hook causes the virtual console to set [hSerialConnectionStatus] to
+; [USING_INTERNAL_CLOCK] ($02) which allows the player to proceed past the
+; Link Receptionists' "Please Wait.".
```
Comment (Rangi42) [inline engine/link/link.asm:2315]:
```suggestion
; This vc_hook causes the Virtual Console to set [hSerialConnectionStatus] to
; USING_INTERNAL_CLOCK, which allows the player to proceed past the
; Link Receptionist's "Please wait."
```

File: engine/menus/save.asm [hunk 4]
```diff
@@ -170,6 +170,8 @@ AddHallOfFameEntry:
 		"sMobileEventIndex is no longer located at 01:be3c."
 	vc_assert BANK(sMobileEventIndexBackup) == $1 && sMobileEventIndexBackup == $be44, \
 		"sMobileEventIndexBackup is no longer located at 01:be44."
+	vc_assert MOBILE_EVENT_OBJECT_GS_BALL == $0b, \
+		"MOBILE_EVENT_OBJECT_GS_BALL is no longer equal to $0b."
```
Comment (Rangi42) [inline engine/menus/save.asm:174]: Please remove the "`($b)`" in the comment.

**Comment by Rangi42** [comment]
Thank you! pokegold will need the `hSerialConnectionStatus` one too.

**Pull request merged**